dialer_test.go

```diff
@@ -0,0 +1,52 @@
+package consul
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/segmentio/objconv/json"
+)
+
+func TestDialer(t *testing.T) {
+	httpServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		res.Write([]byte("Hello World!"))
+	}))
+	defer httpServer.Close()
+	u, _ := url.Parse(httpServer.URL)
+	addr, port := splitHostPort(u.Host)
+
+	consulServer, consulClient := newServerClient(func(res http.ResponseWriter, req *http.Request) {
+		type service struct {
+			Address string
+			Port    int
+		}
+		json.NewEncoder(res).Encode([]struct{ Service service }{{Service: service{addr, port}}})
+	})
+	defer consulServer.Close()
+
+	// The HTTP client uses a transport with a resolver that uses consul to
+	// lookup service addresses.
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			DialContext: (&Dialer{
+				Resolver: &Resolver{
+					Client: consulClient,
+				},
+			}).DialContext,
+		},
+	}
+
+	res, err := httpClient.Get("http://whaetever:0/")
```

Comment:
Here we have to set a port otherwise the http package will assume 80 of 443... I went with zero here but I'm not sure what's ideal.

- always use the value given by consul no matter what the address contains
- use some other value like `:auto` 
- keep it as it is
- ???